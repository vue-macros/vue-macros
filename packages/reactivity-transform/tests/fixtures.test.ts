import { resolve } from 'node:path'
import {
  rollupBuild,
  RollupVue,
  testFixtures,
  UnpluginOxc,
} from '@vue-macros/test-utils'
import { describe } from 'vitest'
import VueReactivityTransform from '../src/rollup'

describe('fixtures', async () => {
  await testFixtures(
    'tests/fixtures/**/*.{vue,js,ts}',
    async (args, id) =>
      rollupBuild(id, [
        VueReactivityTransform(),
        RollupVue({ compiler: await import('vue/compiler-sfc') }),
        UnpluginOxc.rollup(),
      ]),
    {
      cwd: resolve(__dirname, '..'),
      promise: true,
    },
  )
})
