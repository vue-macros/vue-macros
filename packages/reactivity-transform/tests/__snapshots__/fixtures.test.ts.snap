// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fixtures > tests/fixtures/alias.vue 1`] = `
"// alias.js
import { toRef, toDisplayString } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

const _sfc_main = {
  __name: 'alias',
  props: {
  foo: String,
  baz: Number,
},
  setup(__props, { expose: __expose }) {
  __expose();
const propAlias2 = toRef(__props, "baz");
const propAlias = toRef(__props, "foo");



const __returned__ = { propAlias2, propAlias, __MACROS_toRef: toRef };
Object.defineProperty(__returned__, '__isScriptSetup', { enumerable: false, value: true });
return __returned__
}

};

function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return toDisplayString($setup.propAlias) + " " + toDisplayString($setup.propAlias2)
}
var alias = /*#__PURE__*/_export_sfc(_sfc_main, [['render',_sfc_render],__FILE__]);

export { alias as default };
"
`;

exports[`fixtures > tests/fixtures/as.ts 1`] = `
"// as.js
import { ref } from 'vue';

let foo = ref("msg");
foo.value = "world";
console.log(foo);

export { foo };
"
`;

exports[`fixtures > tests/fixtures/basic.ts 1`] = `
"// basic.js
import { ref } from 'vue';

let foo = ref("msg");
foo.value = "world";

export { foo };
"
`;

exports[`fixtures > tests/fixtures/basic.vue 1`] = `
"// basic.js
import { defineComponent, ref } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
	__name: "basic",
	setup(__props, { expose: __expose }) {
		__expose();
		let foo = ref("msg");
		foo.value = "world";
		const __returned__ = {
			get foo() {
				return foo;
			},
			set foo(v) {
				foo = v;
			},
			_ref: ref
		};
		Object.defineProperty(__returned__, "__isScriptSetup", {
			enumerable: false,
			value: true
		});
		return __returned__;
	}
});

var basic = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { basic as default };
"
`;

exports[`fixtures > tests/fixtures/default-value.vue 1`] = `
"// default-value.js
import { defineComponent } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
	__name: "default-value",
	props: {
		fn: {
			type: Function,
			required: true,
			default: () => {
				console.log("hello");
			}
		},
		obj: {
			type: Object,
			required: true,
			default: () => ({ foo: "bar" })
		},
		arr: {
			type: Array,
			required: true,
			default: () => ["foo", "bar"]
		},
		unknown: {
			type: null,
			required: true,
			default: unknownValue
		}
	},
	setup(__props, { expose: __expose }) {
		__expose();
		console.log(__props.fn, __props.obj, __props.arr, __props.unknown);
		const __returned__ = {};
		Object.defineProperty(__returned__, "__isScriptSetup", {
			enumerable: false,
			value: true
		});
		return __returned__;
	}
});

var defaultValue = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { defaultValue as default };
"
`;

exports[`fixtures > tests/fixtures/destructure-props.vue 1`] = `
"// destructure-props.js
import { defineComponent } from 'vue';
import { createPropsRestProxy } from '/vue-macros/reactivity-transform/helper';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
	__name: "destructure-props",
	props: {
		foo: {
			type: String,
			required: true,
			default: "foo"
		},
		bar: {
			type: Number,
			required: true,
			default: 10
		},
		baz: {
			type: Boolean,
			required: true
		}
	},
	setup(__props, { expose: __expose }) {
		__expose();
		const props = createPropsRestProxy(__props, ["foo", "bar"]);
		console.log(__props.foo, __props.bar, props);
		const __returned__ = {
			props,
			get createPropsRestProxy() {
				return createPropsRestProxy;
			}
		};
		Object.defineProperty(__returned__, "__isScriptSetup", {
			enumerable: false,
			value: true
		});
		return __returned__;
	}
});

var destructureProps = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { destructureProps as default };
"
`;

exports[`fixtures > tests/fixtures/issue-608.vue 1`] = `
"// issue-608.js
import { defineComponent, ref, isRef } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
	__name: "issue-608",
	setup(__props, { expose: __expose }) {
		__expose();
		let ref$1 = ref("value");
		console.log(isRef(ref$1.value));
		try {} catch {
			const ref = "";
		}
		console.log(isRef(ref$1.value));
		const __returned__ = {
			get ref() {
				return ref$1;
			},
			set ref(v) {
				ref$1 = v;
			},
			_ref: ref,
			isRef
		};
		Object.defineProperty(__returned__, "__isScriptSetup", {
			enumerable: false,
			value: true
		});
		return __returned__;
	}
});

var issue608 = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { issue608 as default };
"
`;

exports[`fixtures > tests/fixtures/issue-764.ts 1`] = `
"// issue-764.js
import { ref } from 'vue';

let foo = ref("msg");
foo.value = "world";
class LogEntry {}

export { LogEntry };
"
`;

exports[`fixtures > tests/fixtures/runtime-default.vue 1`] = `
"// runtime-default.js
import { defineComponent, mergeDefaults } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
	__name: "runtime-default",
	props: mergeDefaults(["foo"], { foo: "str" }),
	setup(__props, { expose: __expose }) {
		__expose();
		console.log(__props.foo);
		const __returned__ = { __MACROS_mergeDefaults: mergeDefaults };
		Object.defineProperty(__returned__, "__isScriptSetup", {
			enumerable: false,
			value: true
		});
		return __returned__;
	}
});

var runtimeDefault = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { runtimeDefault as default };
"
`;

exports[`fixtures > tests/fixtures/script-and-setup.vue 1`] = `
"// script-and-setup.js
import { ref, defineComponent, createElementBlock, openBlock, toDisplayString } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

let foo = ref("msg");
foo.value = "world";
var _sfc_main = /* @__PURE__ */ defineComponent({
	__name: "script-and-setup",
	setup(__props, { expose: __expose }) {
		__expose();
		foo.value = "foo";
		const __returned__ = {
			get foo() {
				return foo;
			},
			set foo(v) {
				foo = v;
			}
		};
		Object.defineProperty(__returned__, "__isScriptSetup", {
			enumerable: false,
			value: true
		});
		return __returned__;
	}
});

function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock(
    "div",
    null,
    toDisplayString($setup.foo),
    1
    /* TEXT */
  );
}
var scriptAndSetup = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render], __FILE__]);

export { scriptAndSetup as default };
"
`;

exports[`fixtures > tests/fixtures/trailing-comma.vue 1`] = `
"// trailing-comma.js
import { defineComponent, ref } from 'vue';
import _export_sfc from '[NULL]/plugin-vue/export-helper';

var _sfc_main = /* @__PURE__ */ defineComponent({
	__name: "trailing-comma",
	setup(__props, { expose: __expose }) {
		__expose();
		const foo = ref("msg");
		const bar = foo.value;
		const baz = bar;
		const __returned__ = {
			foo,
			bar,
			baz,
			_ref: ref
		};
		Object.defineProperty(__returned__, "__isScriptSetup", {
			enumerable: false,
			value: true
		});
		return __returned__;
	}
});

var trailingComma = /* @__PURE__ */ _export_sfc(_sfc_main, [__FILE__]);

export { trailingComma as default };
"
`;
